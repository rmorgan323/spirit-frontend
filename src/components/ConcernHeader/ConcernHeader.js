/*eslint-disable camelcase*/

import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import { PropTypes } from 'prop-types';
import './ConcernHeader.css';

const ConcernHeader = props => {
  let display;
  const {
    id,
    domain_1,
    domain_2,
    domain_3,
    domain_4,
    domain_5,
    domain_6,
    updated_at
  } = props.selectedConcern;

  if (id !== undefined) {
    display = (
      <div className="concern-header-holder">
        <h4>
          CONCERN ID:<span className="concern-header-span">{id}</span>
        </h4>
        <div className="domain-holder">
          <span className={domain_1 === false ? 'display-none' : null}>1</span>
          <span className={domain_2 === false ? 'display-none' : null}>2</span>
          <span className={domain_3 === false ? 'display-none' : null}>3</span>
          <span className={domain_4 === false ? 'display-none' : null}>4</span>
          <span className={domain_5 === false ? 'display-none' : null}>5</span>
          <span className={domain_6 === false ? 'display-none' : null}>6</span>
        </div>
        <h4>
          UPDATED:<span className="concern-header-span">
            {moment(updated_at).format('LL')}
          </span>
        </h4>
      </div>
    );
  }

  return <div className="ConcernHeader">{display}</div>;
};

const mapStateToProps = store => ({
  selectedConcern: store.selectedConcern
});

export default connect(mapStateToProps, null)(ConcernHeader);

ConcernHeader.propTypes = {
  selectedConcern: PropTypes.object
};
