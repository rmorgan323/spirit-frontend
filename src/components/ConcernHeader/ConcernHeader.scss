@import '../../variables.scss';

.ConcernHeader {
  background-color: $grey1;

  .concern-header-holder {
    display: flex;
    align-items: center;
    padding: 0px 20px 8px;
  }

  .domain-holder {
    display: flex;
    align-items: center;
    margin-right: 20px;

  }

  .domain-holder span {
    padding: 5px 10px;
    background-color: $purple3;
    border: 5px solid $purple5;
    border-radius: 100px;
    font-weight: bold;
    color: $purple1;
    margin-right: 10px;
  }

  .display-none {
    display: none;
  }

  h4 {
    font-size: 14px;
    margin: 0;
    color: $purple4;
  }

  p {
    margin: 0;
    font-size: 16px;
  }

  .concern-header-span {
    color: $purple6;
    font-size: 18px;
    margin: 0 30px 0 10px;
  }
} 