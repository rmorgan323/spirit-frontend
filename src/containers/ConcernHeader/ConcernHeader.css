/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 16px;
  font-family: "Lato", sans-serif;
  outline: none;
  color: #6d6e71; }

h1, h2, h3, h4, h5, h6 {
  font-family: "Josefin Sans", sans-serif;
  line-height: 1.3; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 30px; }

h3 {
  font-size: 24px; }

h4 {
  font-size: 18px; }

h5 {
  font-size: 14px; }

h6 {
  font-size: 10px; }

p {
  font-size: 14px;
  line-height: 1.3;
  font-weight: 900; }

::placeholder {
  opacity: .5;
  text-align: center; }

/***  MIXINS  ***/
.ConcernHeader {
  background-color: #5E378C; }
  .ConcernHeader .concern-header-holder {
    display: flex;
    align-items: center;
    padding: 0px 20px 8px; }
  .ConcernHeader .domain-holder {
    display: flex;
    align-items: center;
    margin-right: 20px; }
  .ConcernHeader .domain-holder span {
    padding: 5px 10px;
    background-color: #BFAFD0;
    border: 5px solid #7F60A4;
    border-radius: 100px;
    font-weight: bold;
    color: #EAE5F0;
    margin-right: 10px; }
  .ConcernHeader .display-none {
    display: none; }
  .ConcernHeader h4 {
    font-size: 14px;
    color: #9E87BA;
    margin: 0 15px 0 0;
    max-width: 455px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap; }
  .ConcernHeader p {
    margin: 0;
    font-size: 16px; }
  .ConcernHeader .concern-header-span {
    color: #EAE5F0;
    font-size: 18px;
    margin: 0 30px 0 10px; }
