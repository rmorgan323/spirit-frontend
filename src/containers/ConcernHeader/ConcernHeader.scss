@import '../../variables.scss';

.ConcernHeader {
  background-color: $purple6;

  .concern-header-holder {
    display: flex;
    align-items: center;
    padding: 0px 20px 8px;
  }

  .domain-holder {
    display: flex;
    align-items: center;
    margin-right: 20px;

  }

  .domain-holder span {
    padding: 5px 10px;
    background-color: $purple3;
    border: 5px solid $purple5;
    border-radius: 100px;
    font-weight: bold;
    color: $purple1;
    margin-right: 10px;
  }

  .display-none {
    display: none;
  }

  h4 {
    font-size: 14px;
    color: $purple4;
    margin: 0 15px 0 0;
    max-width: 455px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  p {
    margin: 0;
    font-size: 16px;
  }

  .concern-header-span {
    color: $purple1;
    font-size: 18px;
    margin: 0 30px 0 10px;
  }
} 