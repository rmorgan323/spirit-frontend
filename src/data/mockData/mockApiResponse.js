const mockApiResponse = {
  status: 'success'
};

export default mockApiResponse;