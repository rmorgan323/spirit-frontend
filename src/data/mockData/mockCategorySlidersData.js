/*eslint-disable camelcase*/

export const mockTitle = 'Capacities';

export const mockSlidersArray = [
  'Attention',
  'Action',
  'Affect',
  'Autonomic',
  'Arousability'
];

export const mockDatabaseNamesArray = [
  'mod_2_attention',
  'mod_2_action',
  'mod_2_affect',
  'mod_2_autonomic',
  'mod_2_arousability'
];

export const mockReminderAsterisk = [null, '*', '*', '*', '*'];

export const mockReminder = '* If selected, remember to make a hypothesized arousal chart.';

export const mockSliderTitle = 'Attention';

export const mockDatabaseName = 'mod_2_attention';

