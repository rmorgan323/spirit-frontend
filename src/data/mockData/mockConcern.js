/*eslint-disable camelcase*/

export const mockConcern = {
  id: 5,
  description: "Concern 1",
  domain_1: true,
  domain_2: true,
  domain_3: false,
  domain_4: true,
  domain_5: false,
  domain_6: false,
  notes: "Some notes.",
  patient_id: 5,
  created_at: "2018-02-20T20:08:19.142Z",
  updated_at: "2018-02-20T20:08:19.142Z"
};
