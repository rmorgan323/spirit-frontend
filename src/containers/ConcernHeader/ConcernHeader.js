/*eslint-disable camelcase*/

import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import moment from 'moment';
import './ConcernHeader.css';

export const ConcernHeader = props => {
  let display;
  const {
    id,
    description,
    domain_1,
    domain_2,
    domain_3,
    domain_4,
    domain_5,
    domain_6,
    updated_at
  } = props.selectedConcern;

  if (id !== undefined) {
    display = (
      <div className="concern-header-holder">
        <h4>
          CONCERN:<span className="concern-header-span">{description}</span>
        </h4>
        <div className="domain-holder">
          <span className={domain_1 === false ? 'display-none' : null}>1</span>
          <span className={domain_2 === false ? 'display-none' : null}>2</span>
          <span className={domain_3 === false ? 'display-none' : null}>3</span>
          <span className={domain_4 === false ? 'display-none' : null}>4</span>
          <span className={domain_5 === false ? 'display-none' : null}>5</span>
          <span className={domain_6 === false ? 'display-none' : null}>6</span>
        </div>
        <h4>
          UPDATED:<span className="concern-header-span">
            {moment(updated_at).format('LL')}
          </span>
        </h4>
      </div>
    );
  }

  return <div className="ConcernHeader">{display}</div>;
};

export const mapStateToProps = store => ({
  selectedConcern: store.selectedConcern
});

export default connect(mapStateToProps, null)(ConcernHeader);

ConcernHeader.propTypes = {
  selectedConcern: PropTypes.object
};
