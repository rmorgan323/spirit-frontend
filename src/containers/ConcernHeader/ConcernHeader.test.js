import React from 'react';
import { ConcernHeader, mapStateToProps } from './ConcernHeader';
import { shallow } from 'enzyme';
import { mockConcern } from '../../data/mockData/mockConcern';

describe('ConcernHeader tests', () => {
  let selectedConcern;
  let renderedConcernHeader;

  beforeEach(() => {
    selectedConcern = mockConcern;

    renderedConcernHeader = shallow(
      <ConcernHeader selectedConcern={selectedConcern} />
    );
  });

  it('Should match the snapshot', () => {
    expect(renderedConcernHeader).toMatchSnapshot();
  });
});

describe('mapStateToProps tests', () => {
  it('Should pull selectedConcern from store', () => {
    const selectedConcern = mockConcern;
    const mockStore = { selectedConcern };
    const result = mapStateToProps(mockStore);

    expect(result.selectedConcern).toEqual(mockStore.selectedConcern);
  });
});
