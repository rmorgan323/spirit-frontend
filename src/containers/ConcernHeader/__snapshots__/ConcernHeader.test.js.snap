// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConcernHeader tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    selectedConcern={
      Object {
        "created_at": "2018-02-20T20:08:19.142Z",
        "description": "Concern 1",
        "domain_1": true,
        "domain_2": true,
        "domain_3": false,
        "domain_4": true,
        "domain_5": false,
        "domain_6": false,
        "id": 5,
        "notes": "Some notes.",
        "patient_id": 5,
        "updated_at": "2018-02-20T20:08:19.142Z",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="concern-header-holder"
      >
        <h4>
          CONCERN:
          <span
            className="concern-header-span"
          >
            Concern 1
          </span>
        </h4>
        <div
          className="domain-holder"
        >
          <span
            className={null}
          >
            1
          </span>
          <span
            className={null}
          >
            2
          </span>
          <span
            className="display-none"
          >
            3
          </span>
          <span
            className={null}
          >
            4
          </span>
          <span
            className="display-none"
          >
            5
          </span>
          <span
            className="display-none"
          >
            6
          </span>
        </div>
        <h4>
          UPDATED:
          <span
            className="concern-header-span"
          >
            February 20, 2018
          </span>
        </h4>
      </div>,
      "className": "ConcernHeader",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            CONCERN:
            <span
              className="concern-header-span"
            >
              Concern 1
            </span>
          </h4>,
          <div
            className="domain-holder"
          >
            <span
              className={null}
            >
              1
            </span>
            <span
              className={null}
            >
              2
            </span>
            <span
              className="display-none"
            >
              3
            </span>
            <span
              className={null}
            >
              4
            </span>
            <span
              className="display-none"
            >
              5
            </span>
            <span
              className="display-none"
            >
              6
            </span>
          </div>,
          <h4>
            UPDATED:
            <span
              className="concern-header-span"
            >
              February 20, 2018
            </span>
          </h4>,
        ],
        "className": "concern-header-holder",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "CONCERN:",
              <span
                className="concern-header-span"
              >
                Concern 1
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "CONCERN:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Concern 1",
                "className": "concern-header-span",
              },
              "ref": null,
              "rendered": "Concern 1",
              "type": "span",
            },
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className={null}
              >
                1
              </span>,
              <span
                className={null}
              >
                2
              </span>,
              <span
                className="display-none"
              >
                3
              </span>,
              <span
                className={null}
              >
                4
              </span>,
              <span
                className="display-none"
              >
                5
              </span>,
              <span
                className="display-none"
              >
                6
              </span>,
            ],
            "className": "domain-holder",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1",
                "className": null,
              },
              "ref": null,
              "rendered": "1",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2",
                "className": null,
              },
              "ref": null,
              "rendered": "2",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "3",
                "className": "display-none",
              },
              "ref": null,
              "rendered": "3",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "4",
                "className": null,
              },
              "ref": null,
              "rendered": "4",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "5",
                "className": "display-none",
              },
              "ref": null,
              "rendered": "5",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "6",
                "className": "display-none",
              },
              "ref": null,
              "rendered": "6",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "UPDATED:",
              <span
                className="concern-header-span"
              >
                February 20, 2018
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "UPDATED:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "February 20, 2018",
                "className": "concern-header-span",
              },
              "ref": null,
              "rendered": "February 20, 2018",
              "type": "span",
            },
          ],
          "type": "h4",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="concern-header-holder"
        >
          <h4>
            CONCERN:
            <span
              className="concern-header-span"
            >
              Concern 1
            </span>
          </h4>
          <div
            className="domain-holder"
          >
            <span
              className={null}
            >
              1
            </span>
            <span
              className={null}
            >
              2
            </span>
            <span
              className="display-none"
            >
              3
            </span>
            <span
              className={null}
            >
              4
            </span>
            <span
              className="display-none"
            >
              5
            </span>
            <span
              className="display-none"
            >
              6
            </span>
          </div>
          <h4>
            UPDATED:
            <span
              className="concern-header-span"
            >
              February 20, 2018
            </span>
          </h4>
        </div>,
        "className": "ConcernHeader",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              CONCERN:
              <span
                className="concern-header-span"
              >
                Concern 1
              </span>
            </h4>,
            <div
              className="domain-holder"
            >
              <span
                className={null}
              >
                1
              </span>
              <span
                className={null}
              >
                2
              </span>
              <span
                className="display-none"
              >
                3
              </span>
              <span
                className={null}
              >
                4
              </span>
              <span
                className="display-none"
              >
                5
              </span>
              <span
                className="display-none"
              >
                6
              </span>
            </div>,
            <h4>
              UPDATED:
              <span
                className="concern-header-span"
              >
                February 20, 2018
              </span>
            </h4>,
          ],
          "className": "concern-header-holder",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "CONCERN:",
                <span
                  className="concern-header-span"
                >
                  Concern 1
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "CONCERN:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Concern 1",
                  "className": "concern-header-span",
                },
                "ref": null,
                "rendered": "Concern 1",
                "type": "span",
              },
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className={null}
                >
                  1
                </span>,
                <span
                  className={null}
                >
                  2
                </span>,
                <span
                  className="display-none"
                >
                  3
                </span>,
                <span
                  className={null}
                >
                  4
                </span>,
                <span
                  className="display-none"
                >
                  5
                </span>,
                <span
                  className="display-none"
                >
                  6
                </span>,
              ],
              "className": "domain-holder",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                  "className": null,
                },
                "ref": null,
                "rendered": "1",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2",
                  "className": null,
                },
                "ref": null,
                "rendered": "2",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "3",
                  "className": "display-none",
                },
                "ref": null,
                "rendered": "3",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "4",
                  "className": null,
                },
                "ref": null,
                "rendered": "4",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "5",
                  "className": "display-none",
                },
                "ref": null,
                "rendered": "5",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "6",
                  "className": "display-none",
                },
                "ref": null,
                "rendered": "6",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "UPDATED:",
                <span
                  className="concern-header-span"
                >
                  February 20, 2018
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "UPDATED:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "February 20, 2018",
                  "className": "concern-header-span",
                },
                "ref": null,
                "rendered": "February 20, 2018",
                "type": "span",
              },
            ],
            "type": "h4",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
